### DISCUSSION TITLE
Lick debounce

### DISCUSSION THREAD

**Discussion by AtMostafa** [pyControl/code#105] [Q&A] [answered]
Hi,

The lick detector (PyControl hardware) detects a single lick _sometimes_ when the reward solenoid (triggered by the same circuit) is triggered. The debounce value is the default 5ms. Should I increase it? Does anyone have a similar experience? Is it recommended to add a capacitor or something to the circuit to dampen possible electrical spikes (not action potentials😄)?

**Comment by ThomasAkam** [comment, marked as answer]
Hi @AtMostafa,  I don't think debouncing is going to help with this because it prevents multiple events from being generated within a specified window, but will not stop a single event being generated due to crosstalk between the solenoid and lick detector.  The best way to resolve this problem is to minimise electrical coupling between the solenoid circuitry and the sensitive part of the lick detection circuitry, which includes the lick spout itself and the wire connecting it to the lickometer.  The main ways to achieve this are:
1. Increase the physical distance between the circuits, e.g. by having the solenoid plugged into a seperate port adapter board rather than into the lickometer board.  Also, ensure that the wire that goes to the lick spout is well away from the solenoid and associated wiring.
2. Shielding the lick detection circuit using a conductor (e.g. aluminium foil) that is connected to the pyControl ground.

Some combination of these steps should fix the issue.  Let me know how you get on.

**Reply by AtMostafa** [reply]
Alright. I am using the solenoid controller on the lick detector board itself right now. I will move it to another port and try your suggestions. Many thanks!

**Reply by AtMostafa** [reply]
Update: this did work.
